### PR TITLE
[AERIE-1940]  scheduler populate source metadata field

### PR DESCRIPTION
**Pull request by dyst5422**
* **Tickets addressed:** AERIE-1940
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

Adjusted the GraphQL queries to use parameterized queries. Built a map of goals to activities, used that map to store the goal id for each activity that had a source goal.

## Verification

Tested locally.

## Documentation

None

## Future work

There are a bunch more graphql queries that could really use some cleaning up.

### CODE REVIEW

**Comment by dyst5422** [comment]
@mattdailis and @adrienmaillard I think you two are most familiar with this section of the codebase and so you both drew the lucky straws of reviewing this PR.

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java [hunk 1]
```diff
@@ -106,6 +108,41 @@ private Optional<JsonObject> postRequest(final String gqlStr) throws IOException
...
+      if (httpResp.statusCode() != 200) {
+        //TODO: how severely to error out if aerie cannot be reached or has a 500 error or json is garbled etc etc?
+        return Optional.empty();
+      }
+      final var respBody = Json.createReader(httpResp.body()).readObject();
+      if (respBody.containsKey("errors")) {
+        throw new PlanServiceException(respBody.toString());
+      }
+      return Optional.of(respBody);
+    } catch (final InterruptedException e) {
```
Comment (mattdailis) [inline scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java:138]: We should at least reset the interrupted flag here (Thread.currentThread().interrupt())

My limited understanding is that InterruptedException is thrown when the thread receives a request to terminate (ctrl+c or similar...?), so I don't think we'll want to retry in that scenario.

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java [hunk 2]
```diff
@@ -114,9 +151,16 @@ private Optional<JsonObject> postRequest(final String gqlStr) throws IOException
...
+    final var query = """
+        query GetPlanRevision($id: Int!) {
+          plan_by_pk(id: $id) {
+            revision
+          }
+        }
+        """;
+    final var variables = Json.createObjectBuilder().add("id", planId.id()).build();
+
+    final var response = postRequest(query, variables).orElseThrow(() -> new NoSuchPlanException(planId));
```
Comment (mattdailis) [inline scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java:163]: I believe this was already the case before this PR, so I won't insist on fixing it now, but I think that throwing NoSuchPlanException every time anything goes wrong is misleading and makes it difficult to determine what actually went wrong.

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java (hunk 1 above)
Comment (dyst5422) [reply scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java:138]: I 100% copy pasted the new request with arguments method from the existing request by string method, so I will 100% defer to you on this element and happy to make the change in both methods

File: scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java (hunk 2 above)
Comment (dyst5422) [reply scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java:163]: I think in this case, it is only when the postRequest returns an empty optional, in which case it "should" interpret as the plan doesn't exist. I think the issue is in our liberal returning of Optional.empty in the postRequest methods for cases that are not actually an "empty" return value, but an error that occurred during resolution of the response (ie non 200 status codes, and interruptions)

Comment (dyst5422) [reply scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java:163]: Generally operating fast and loose with failure modes

**Comment by dyst5422** [comment]
@mattdailis since both of the above concerns were pre-existing, how do we feel about merging this in and addressing them in a tech-debt ticket I can slot in for this coming release?

**Review by mattdailis — APPROVED** [review]

**Review by goetzrrGit — APPROVED** [review]
I don't see anything wrong. I like how you clean up the gql calls. They are more readable now. 

Just make sure you have a ticket in JIRA to go back and update the `TODO`

**Pull request merged**